org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/versionrange/AbstractVersionRangeAssert.java

```diff
@@ -0,0 +1,278 @@
+		isNotNull();
+		if (actual.getRightType() != RIGHT_OPEN) {
+			failWithMessage("%nExpecting version range%n <%s>%nto have open upper bound%nbut it was closed", actual);
+		}
+		return myself;
+	}
+
+	public SELF isRightClosed() {
+		isNotNull();
+		if (actual.getRightType() != RIGHT_CLOSED) {
+			failWithMessage("%nExpecting version range%n <%s>%nto have closed upper bound%nbut it was open", actual);
+		}
+		return myself;
+	}
+
+	public SELF includes(String version) {
+		return includes(Version.valueOf(version));
+	}
+
+	public SELF includes(Version version) {
+		isNotNull();
+		if (!actual.includes(version)) {
+			failWithMessage("%nExpecting version range%n <%s>%nto include version%n <%s>%nbut it does not", actual,
+				version);
+		}
+		return myself;
+	}
+
+	public SELF doesNotInclude(String version) {
+		return doesNotInclude(Version.valueOf(version));
+	}
+
+	public SELF doesNotInclude(Version version) {
+		isNotNull();
+		if (actual.includes(version)) {
+			failWithMessage("%nExpecting version range%n <%s>%nto not include version%n <%s>%nbut it does", actual,
+				version);
+		}
+		return myself;
+	}
+
+	public SELF isEmpty() {
+		isNotNull();
+		if (!actual.isEmpty()) {
+			failWithMessage("%nExpecting version range%n <%s>%nto be empty%nbut it was not", actual);
+		}
+		return myself;
+	}
+
+	public SELF isNotEmpty() {
+		isNotNull();
+		if (actual.isEmpty()) {
+			failWithMessage("%nExpecting version range%n <%s>%nto be empty%nbut it was not", actual);
+		}
+		return myself;
+	}
+
+	public SELF isExact() {
+		isNotNull();
+		if (!actual.isExact()) {
+			failWithMessage("%nExpecting version range%n <%s>%nto be exact%nbut it was not", actual);
+		}
+		return myself;
+	}
+
+	public SELF isNotExact() {
+		isNotNull();
+		if (actual.isExact()) {
+			failWithMessage("%nExpecting version range%n <%s>%nto be exact%nbut it was not", actual);
+		}
+		return myself;
+	}
+
+	public SELF intersects(String... version) {
+		return intersects(Stream.of(version)
+			.map(VersionRange::valueOf)
+			.toArray(VersionRange[]::new));
+	}
+
+	public SELF intersects(VersionRange... versions) {
+		isNotNull();
+		if (actual.intersection(versions)
+			.isEmpty()) {
+			failWithMessage("%nExpecting version range%n <%s>%nto intersect all of%n <%s>%nbut it does not", actual,
+				Arrays.toString(versions));
+		}
+		return myself;
+	}
+
+	public SELF doesNotIntersect(String... versions) {
+		return doesNotIntersect(Stream.of(versions)
+			.map(VersionRange::valueOf)
+			.toArray(VersionRange[]::new));
+	}
+
+	public SELF doesNotIntersect(VersionRange... versions) {
+		isNotNull();
+		VersionRange intersection = actual.intersection(versions);
+		if (!intersection
+			.isEmpty()) {
+			failWithMessage(
+				"%nExpecting version range%n <%s>%nto not intersect any of%n <%s>%nbut it has intersection%n <%s>",
+				actual, Arrays.toString(versions), intersection);
+		}
+		return myself;
+	}
+
+	// public AbstractIntegerAssert<?> hasMinorThat() {
```

Comment:
I don't think the commented section should be in the PR. There is some overlap (isEmpty) with the code.

Reply:
Oops. Done.